### ISSUE TITLE
[Quiz Creation] - Update `ExamSerializer` to validate `question_sources` v3

### ISSUE THREAD

**Issue by nucleogenesis** [learningequality/kolibri#11580]
## Overview

The `ExamSerializer` needs to be updated accordingly so that we can save the new version of the `Exam`. 

## Details

#11002 updated the `Exam` model with a new version for the `question_sources` property. Previously, the `question_sources` property was where the quiz associated exercises were listed. Now `question_sources` instead refers to the newly added "Quiz Sections" -- each of which then contains their own list of exercises.

The existing `QuestionSourceSerializer`, which is used to serialize each item in the `question_sources` , will likely need to be renamed to something like `QuestionSerializer` and the `QuestionSourceSerializer` name should instead validate the shape of a quiz section:

```python
# excerpted from the comment block in kolibri/core/exams/models.py#Exam~L63:
{
      "section_id": <a uuid unique to this section>,
      "section_title": <section title>,
      "description": <section description>,
      "resource_pool": [ <contentnode_ids of pool of resources> ],
      "question_count": <number of questions in section>,
      "learners_see_fixed_order": <bool>,
      "questions": [
        {
            "exercise_id": <exercise_pk>,
            "question_id": <item_id_within_exercise>,
            "title": <title of question>,
            "counter_in_exercise": <unique_count_for_question>,
        },
      ]
},
```

Then the `QuestionSerializer` can be used similarly to then validate each child of the `questions` property defined on the serializer that handles the above data structure.

Note that the client-side _should_ work as-is but you may need to look at the `saveQuiz` method in `useQuizCreation` if something about the data passed to that API call is off.

## Acceptance criteria

- [ ] When you click "Save" in quiz creation, the quiz is saved to the database and validated properly.

**Comment by nick2432** [comment]
Can I contribute to this issue? I'm new here, but I have experience contributing to many other organizations. Is there a Discord channel or any other channel where I can discuss and get assistance?

**Comment by ozer550** [comment]
Hi @nick2432! Thank you for reaching out to contribute to this issue, although I have already started working on it. You may find other open issues in the repository that could interest you! Regarding any questions or assistance related with any issue, you could just directly comment on it or you could also refer to [this](https://github.com/learningequality/kolibri/discussions/10691).

**Comment by nucleogenesis** [comment]
This is closed by #11674
